### PR TITLE
Add disable and visible condition to TypeToolbarAction

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Add disable and visible condition to TypeToolbarAction.

#### Why?

Allows to hide or disable TypeToolbarAction.

#### To Do

- [ ] Create a documentation PR
- [ ] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/TypeToolbarAction.test.js [hunk 1]
```diff
@@ -130,3 +131,29 @@ test('Throw error if no types are available in FormStore', () => {
...
+});
+
+test('Return item config when passed visible condition is met', () => {
+    const typeToolbarAction = createTypeToolbarAction({visible_condition: 'url == "/"'});
+    typeToolbarAction.resourceFormStore.data.url = '/';
+
+    expect(typeToolbarAction.getToolbarItemConfig()).toBeDefined();
+});
+
+test('Return disabled true when passed disable condition is not met', () => {
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/tests/toolbarActions/TypeToolbarAction.test.js]: `... disabled false`

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js [hunk 2]
```diff
@@ -10,6 +11,20 @@ export default class TypeToolbarAction extends AbstractFormToolbarAction {
             throw new Error('The ToolbarAction for types only works with entities actually supporting types!');
         }
 
+        const {
+            visible_condition: visibleCondition,
+            disable_condition: disableCondition,
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js]: i would probably use `disabled_condition` to match the forms 🙂

**Comment by nnatter** [comment]
should we use the `disabled_condition` in the `PageAdmin`?

**Comment by alexander-schranz** [comment]
@nnatter PageAdmin doesn't use the TypeToolbarAction it seems to have its own TemplateToolbarAction.

File: src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php [hunk 3]
```diff
@@ -118,7 +118,12 @@ public function configureViews(ViewCollection $viewCollection): void
                     'save_visible_condition' => '(!_permissions || _permissions.edit)',
                 ]
             ),
-            new ToolbarAction('sulu_page.templates'),
+            new ToolbarAction(
+                'sulu_page.templates',
+                [
+
```
Comment (nnatter) [inline src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php]: think there is something missing here

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
+        } = this.options;
+
+        if (visibleCondition && !jexl.evalSync(visibleCondition, this.resourceFormStore.data)) {
+            return;
+        }
+
+        let isDisabled = false;
+        if (disabledCondition && jexl.evalSync(disabledCondition, this.resourceFormStore.data)) {
+            isDisabled = true;
+        }
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js]: Would rewrite that to use `const`: 

```javascript
const isDisabled = disabledCondition ? jexl.evalSync(disabledCondition, this.resourceFormStore.data) : false;
```

File: src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js [hunk 5]
```diff
@@ -30,6 +31,20 @@ export default class TemplateToolbarAction extends AbstractFormToolbarAction {
             throw new Error('The ToolbarAction for types only works with entities actually supporting types!');
         }
 
+        const {
+            visible_condition: visibleCondition,
```
Comment (danrot) [inline src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js]: This `ToolbarAction` is only usable in the `PageAdmin`, and there we don't need this property, right? Then I would not implement it as well.

File: src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js [hunk 6]
```diff
@@ -30,6 +31,20 @@ export default class TemplateToolbarAction extends AbstractFormToolbarAction {
...
+        } = this.options;
+
+        if (visibleCondition && !jexl.evalSync(visibleCondition, this.resourceFormStore.data)) {
+            return;
+        }
+
+        let isDisabled = false;
+        if (disabledCondition && jexl.evalSync(disabledCondition, this.resourceFormStore.data)) {
+            isDisabled = true;
+        }
```
Comment (danrot) [inline src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js]: Same as above.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js [hunk 7]
```diff
@@ -1,15 +1,30 @@
...
-    getToolbarItemConfig(): ToolbarItemConfig {
+    getToolbarItemConfig(): ?ToolbarItemConfig {
         const formTypes = this.resourceFormStore.types;
 
         if (!this.resourceFormStore.typesLoading && Object.keys(formTypes).length === 0) {
             throw new Error('The ToolbarAction for types only works with entities actually supporting types!');
         }
 
+        const {
+            visible_condition: visibleCondition,
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/TypeToolbarAction.js]: Actually I am not sure if this one makes sense here at all... If we do not display the type dropdown all together, then people without this `ToolbarAction` won't see what type is selected.

**Comment by alexander-schranz** [comment]
@danrot @nnatter comments fixed.

**Pull request merged**